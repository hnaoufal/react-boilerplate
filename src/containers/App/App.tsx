import * as React from 'react';
import Reviews from '../Reviews/Reviews';
import { Provider } from 'react-redux';
import { store } from '../../store';

import styled from 'styled-components';

const AppContainer = styled.div`
  color: red;
`;

export default class App extends React.Component {
  public render() {
    return (
    <Provider store={store}>
      <AppContainer className="App">
        Hello World
        <Reviews />
      </AppContainer>
    </Provider>
    );
  }
}
