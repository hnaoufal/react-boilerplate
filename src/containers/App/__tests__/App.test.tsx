import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import App from '../App';

import toJson from 'enzyme-to-json';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('should match the Snapshot', () => {
    const element = shallow(<App />);

    expect(toJson(element)).toMatchSnapshot();
  })
});