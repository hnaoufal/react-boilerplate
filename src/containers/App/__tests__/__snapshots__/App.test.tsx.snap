// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the Snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <styled.div
    className="App"
  >
    Hello World
    <Connect(Reviews) />
  </styled.div>
</Provider>
`;
